enterprise/backend/src/metabase_enterprise/database_replication/api.clj

```diff
@@ -28,28 +36,143 @@
 (defn- database-id->connection-id [conns database-id]
   (->> database-id kw-id (get conns) :connection-id))
 
+(defn- schema-filter->fn [{:keys [type patterns]}]
+  (fn [table-schema]
+    (or (= type "all")
+        (let [xform (case type
+                      "include" identity
+                      "exclude" nil?)]
+          (-> patterns
+              driver.s/schema-pattern->re-pattern
+              (re-matches table-schema)
```

Comment:
I just noticed this is doing a lot of work for each table-schema. Maybe a decently bounded set, but maybe this could benefit from a bit of cleanup?

```clojure
(defn- schema-filter->fn [{t :type patterns :patterns}]
  (if (= t "all")
    (constantly true)
    (let [pat (driver.s/schema-pattern->re-pattern patterns)
          f   (case t
                "include" some?
                "exclude" nil?)]
      (fn [table-schema]
        (f (re-matches pat table-schema))))))
```

maybe bikeshedding. not necessary but up to you.